SEED-platform/seed
# Remove Deprecated APIs

#### Any background context you want to provide?
SEED API namespaces `v1`, `v2`, and `v2.1` have been deprecated for a very long time, stopped receiving updates, and are not expected to function as intended.

#### What's this PR do?
- Removes all endpoints associated with deprecated API namespaces
  - Attempting to request anything using `/api/v(1|2|2.1)/*` will result in a `410 GONE` response
- Updates tests that were relying on older endpoints or data structures
- Removes unused methods
- Removes deprecated tests
- Centralizes utility functions (like `get_js_role`)
- Replaces explicit `manage.py test seed` configuration with no target so that `seed` and `config` both run
- Removes `INCLUDE_SEED_V2_APIS` environment variable

#### How should this be manually tested?
- Confirm that old endpoints no longer work
- Verify that API v3 behavior remains unchanged
